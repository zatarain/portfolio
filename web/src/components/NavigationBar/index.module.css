.navigation-bar {
	box-sizing: border-box;
	display: flex;
	align-items: center;
	border-bottom: 1px solid var(--highlight-secondary);
	justify-content: space-between;
	padding: 0 .5rem;
}

.navigation-bar a {
	display: inline-block;
}

.logo:link,
.logo:visited {
	line-height: 100%;
	font-size: 1.5rem;
	font-weight: 600;
	height: 3.4rem;
	padding: 0;
}

.logo img {
	vertical-align: middle;
}

.sections {
	display: flex;
	list-style-type: none;
	padding: 0;
	margin: 0;
	gap: 1rem;
}

.sections li {
	line-height: 2.25rem;
}

.sections :hover {
	color: var(--fg-menuitem-hover);
}

.dropdown {
	display: none;
}

.call-to-action {
	background: var(--bg-action);
	box-shadow: 0px 1px 2px rgba(16, 24, 40, 0.04);
	border-radius: 6px;
	padding: 0 1rem;
}

.call-to-action:hover {
	color: white;
	background: var(--bg-action-hover);
}

@media screen and (max-width: 800px) {
	.navigation-bar {
		display: block;
		z-index: 3;
	}

	.sections li {
		display: none;
	}

	.sections .dropdown {
		display: block;
		font-size: 2.5rem;
		position: absolute;
		right: 0;
		top: 0;
		padding: .5rem .75rem;
	}

	.responsive .sections {
		display: block;
		position: absolute;
		width: 100%;
		right: 0;
		top: 0;
	}

	.responsive .sections li:not(.dropdown, .action) {
		background: var(--bg-secondary);
		display: block;
		padding: 1rem;
		position: relative;
		top: 3.5rem;
	}
}

@media screen and (max-width: 340px) {
	.navigation-bar {
		grid-template-columns: 14ch auto;
	}

	.logo {
		max-width: 12ch;
		overflow: hidden;
		text-overflow: ellipsis;
	}
}
