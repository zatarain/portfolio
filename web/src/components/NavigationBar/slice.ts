import { createSlice } from '@reduxjs/toolkit'
import { AppState } from '#store'

export interface ResponsiveState {
	value: boolean
}

const initialState = { value: false } as ResponsiveState

const slice = createSlice({
	name: 'responsive',
	initialState,
	reducers: {
		flip(state) {
			state.value = !state.value
		},
		close(state) {
			state.value = false
		}
	}
})

export const flipResponsiveMenu = slice.actions.flip
export const closeResponsiveMenu = slice.actions.close
export const selectResponsive = (state: AppState) => state.responsive.value
export default slice.reducer
