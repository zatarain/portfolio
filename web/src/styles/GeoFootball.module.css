.geo-football main {
	margin: 0;
	padding: 0 0 0 0;
	height: 94vh;
}

.geo-football footer {
	margin: 0;
	position: sticky;
	bottom: 0;
}

.geo-football footer ul {
	display: none;
}
