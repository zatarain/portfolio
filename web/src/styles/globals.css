@import url('https://cdnjs.cloudflare.com/ajax/libs/font-awesome/4.7.0/css/font-awesome.min.css');
@import url('https://fonts.cdnfonts.com/css/cascadia-code');
@import url('react-responsive-carousel/lib/styles/carousel.min.css');
@import url('leaflet/dist/leaflet.css');


:root {
  --bg-page: #fafbfc;
  --bg-primary: #28b873;
  --bg-secondary: #151b28;
  --bg-action: #f40;
  --bg-action-hover: #f60;
  --link-secondary: #a5acba;
  --highlight-secondary: #5cdeac;
  --fg-secondary: #fff;
  --fg-menuitem-hover: #7fffd4;
  --ldn-dark: #29bc9b;
  --ldn-lighter: #aaffec;
  --ldn-light: #79ffe1;
  --ldn: #50e3c2;
  --heading-2: var(--ldn-dark);
  --heading-3: var(--ldn-dark);
  --headings-font: Cambria, Cochin, Georgia, Times, 'Times New Roman', serif;
}

@media screen {
  html, body {
    padding: 0;
    margin: 0;
    text-align: center;
    font-family: Inter, BlinkMacSystemFont, Segoe UI, Roboto, Oxygen, Ubuntu,
      Cantarell, Fira Sans, Droid Sans, Helvetica Neue, sans-serif;
    font-size: 14px;
    scroll-behavior: smooth;
  }

  main {
    padding: 0 3rem;
    margin: 3rem 0;
    font-size: 1.2rem;
    line-height: 1.75rem;
  }
}

* {
  box-sizing: border-box;
}

a:link, a:visited {
  color: var(--ldn-dark);
  text-decoration: none;
}

a:hover, a:hover:visited {
  color: var(--bg-action);
}

.carousel ul.control-dots {
	bottom: 4rem;
	z-index: 3;
}

.carousel ul.control-dots li {
  width: 0.5rem;
  height: 0.5rem;
  padding: 0.5rem;
}

.carousel figcaption {
  margin: 0;
  position: relative;
  bottom: 3rem;
  font-size: 1rem;
}

.leaflet-bottom,
.leaflet-control,
.leaflet-pane,
.leaflet-top {
  z-index: 0 !important;
}

@media screen and (max-width: 850px) {
  .carousel figure img {
    height: 50vh;
  }

  .carousel figcaption {
    bottom: 1rem;
  }

  .carousel ul.control-dots {
    bottom: 3rem;
  }
}

@media screen and (max-width: 539px) {
  .carousel figure img {
    height: 30vh;
  }

  .carousel figcaption {
    bottom: 3rem;
    display: none;
  }

  .carousel ul.control-dots {
    bottom: -1rem;
  }
}
