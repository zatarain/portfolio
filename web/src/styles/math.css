@import url('https://cdn.jsdelivr.net/npm/katex@0.16.0/dist/katex.min.css');

@font-face{
  font-family: Euler;
  font-style: normal;
  font-weight: 500;
  src: url(https://fonts.cdnfonts.com/s/19939/euler.woff) format('woff');
}

.katex {
  font: normal 1.21em Euler, 'Cambria Math', KaTeX_Main, 'Times New Roman', serif;
}

/* Text font families. */
.katex .textrm {
  font-family: Euler, 'Cambria Math', KaTeX_Main;
}

.katex .textsf {
  font-family: 'Cambria Math', KaTeX_SansSerif;
}

.katex .texttt {
  font-family: 'Cambria Math', KaTeX_Typewriter;
}

/* Math fonts. */
.katex .mathnormal {
  font-family: Euler, 'Cambria Math', KaTeX_Math;
  font-style: italic;
}

.katex .mathit {
  font-family: Euler, 'Cambria Math', KaTeX_Main;
  font-style: italic;
}

.katex .mathrm {
  font-style: normal;
}

.katex .mathbf {
  font-family: Euler, 'Cambria Math', KaTeX_Main;
  font-weight: bold;
}

.katex .boldsymbol {
  font-family: Euler, 'Cambria Math', KaTeX_Math;
  font-weight: bold;
  font-style: italic;
}

.katex .amsrm {
  font-family: 'Cambria Math', KaTeX_AMS;
}

.katex .mathbb,
.katex .textbb {
  font-family: 'Cambria Math', KaTeX_AMS;
}

.katex .mathcal {
  font-family: 'Cambria Math', KaTeX_Caligraphic;
}

.katex .mathfrak,
.katex .textfrak {
  font-family: 'Cambria Math', KaTeX_Fraktur;
}

.katex .mathtt {
  font-family: 'Cambria Math', KaTeX_Typewriter;
}

.katex .mathscr,
.katex .textscr {
  font-family: 'Cambria Math', KaTeX_Script;
}

.katex .mathsf,
.katex .textsf {
  font-family: 'Cambria Math', KaTeX_SansSerif;
}

.katex .mathboldsf,
.katex .textboldsf {
  font-family: 'Cambria Math', KaTeX_SansSerif;
  font-weight: bold;
}

.katex .mathitsf,
.katex .textitsf {
  font-family: 'Cambria Math', KaTeX_SansSerif;
  font-style: italic;
}

.katex .mainrm {
  font-family: Euler, 'Cambria Math', KaTeX_Main;
  font-style: normal;
}

/* Delimiters: Parenthesis, Brackets, Braces,
	Norm, Absolute Value, Angle Parenthesis, ... */
/**
.katex .delimsizing.size1 {
  font-family: Euler, 'Cambria Math', KaTeX_Size1;
}

.katex .delimsizing.size2 {
  font-family: Euler;
  font-size: 1.5em;
}

.katex .delimsizing.size3 {
  font-family: Euler;
  font-size: 3em;
}

.katex .delimsizing.size4 {
  font-family: Euler;
  font-size: 2.5em;
}

.katex .delimsizing.mult .delim-size1 > span {
  font-family: Euler, 'Cambria Math', KaTeX_Size1;
}

.katex .delimsizing.mult .delim-size4 > span {
  font-family: Euler;
  font-size: 2.5em;
}
/**/

/* Operator Symbols */
.katex .op-symbol.small-op {
  font-family: Euler, 'Cambria Math', KaTeX_Size1;
}

.katex .op-symbol.large-op {
  font-family: Euler;
  font-size: 1.5em;
  bottom: -0.2em!important;
  margin-right: 0!important;
}
