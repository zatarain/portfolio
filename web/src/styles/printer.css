@media print {
	@page {
		margin: 1cm;
	}

	header {
		display: none;
	}

	nav {
		display: none!important;
	}

	footer {
		display: none;
	}

	html, body {
		font-size: 10pt;
    margin: 0;
		padding: 0;
    text-align: center;
    font-family: Arial, Helvetica, sans-serif, BlinkMacSystemFont, Segoe UI, Roboto, Oxygen, Ubuntu,
      Cantarell, Fira Sans, Droid Sans, Helvetica Neue, sans-serif;
 	}
}
