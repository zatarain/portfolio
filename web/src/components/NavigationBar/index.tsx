import Image from 'next/image'
import Link from 'next/link'
import { SyntheticEvent } from 'react'
import { useAppDispatch, useAppSelector } from '#hooks'
import { flipResponsiveMenu, selectResponsive } from './slice'
import styles from './index.module.css'

interface Properties {
	name: string
}

const NavigationBar = ({ name }: Properties) => {
	const dispatch = useAppDispatch()
	const responsive = useAppSelector(selectResponsive)
	const className = responsive ? styles.responsive : ''
	const onOpenMenuClick = (event: SyntheticEvent) => {
		event.preventDefault()
		dispatch(flipResponsiveMenu())
	}

	return (
		<nav id="navigation-bar" className={`${styles['navigation-bar']} ${className}`} role="menubar">
			<Link href="/" className={styles.logo}>
				<Image alt="logo" src="/logo.svg" width="48" height="48" />
				{name}
			</Link>
			<ul className={styles.sections}>
				<li><Link href="#work-experience" scroll={false}>Experience</Link></li>
				<li><Link href="#education" scroll={false}>Education</Link></li>
				<li><Link href="#" scroll={false}>Skills</Link></li>
				<li><Link href="#academic-projects" scroll={false}>Projects</Link></li>
				<li><Link href="#volunteering" scroll={false}>Hobbies</Link></li>
				<li className={styles.action}><Link href="#">E-mail</Link></li>
				<li className={styles.action}>
					<Link href="#" className={styles['call-to-action']}>Download</Link>
				</li>
				<li className={styles.dropdown}>
					<Link href="#" onClick={onOpenMenuClick} role="menuitemcheckbox">
						<i className="fa fa-bars"></i>
					</Link>
				</li>
			</ul>
		</nav>
	)
}

export default NavigationBar
