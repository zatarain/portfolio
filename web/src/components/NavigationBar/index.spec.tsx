import { render } from '@testing-library/react'
import userEvent from '@testing-library/user-event'
import { Provider } from 'react-redux'
import styles from './index.module.css'

import { makeStore } from '#store'
import NavigationBar from '.'

describe('<NavigationBar />', () => {
	it('renders the component correctly with passed name', () => {
		const store = makeStore()

		const { asFragment, getByText } = render(
			<Provider store={store}>
				<NavigationBar name="My Name" />
			</Provider>
		)

		expect(getByText('My Name')).toBeInTheDocument()
		expect(asFragment()).toMatchSnapshot()
	})

	it('adds class name "responsive" when user clicks menu icon', async () => {
		const store = makeStore()
		const user = userEvent.setup()

		const { container } = render(
			<Provider store={store}>
				<NavigationBar name="Test Name" />
			</Provider>
		)
		const dropdown = container.querySelector(`li.${styles.icon} a`)
		expect(dropdown).not.toBeNull()
		const bar = container.querySelector('nav')
		await user.click(dropdown)
		expect(bar).toHaveClass(styles.responsive)
	})
})
