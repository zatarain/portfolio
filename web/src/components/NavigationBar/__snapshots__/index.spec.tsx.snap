// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NavigationBar /> renders the component correctly with passed name 1`] = `
<DocumentFragment>
  <nav
    class="navigation-bar "
    id="navigation-bar"
  >
    <a
      class="logo"
      href="/"
    >
      <img
        alt="logo"
        data-nimg="1"
        decoding="async"
        height="48"
        loading="lazy"
        src="/logo.svg"
        style="color: transparent;"
        width="48"
      />
      My Name
    </a>
    <ul
      class="sections"
    >
      <li>
        <a
          href="#"
        >
          Experience
        </a>
      </li>
      <li>
        <a
          href="#"
        >
          Education
        </a>
      </li>
      <li>
        <a
          href="#"
        >
          Skills
        </a>
      </li>
      <li>
        <a
          href="#"
        >
          Projects
        </a>
      </li>
      <li>
        <a
          href="#"
        >
          Hobbies
        </a>
      </li>
      <li
        class="action"
      >
        <a
          href="#"
        >
          E-mail
        </a>
      </li>
      <li
        class="action"
      >
        <a
          class="call-to-action"
          href="#"
        >
          Download
        </a>
      </li>
      <li
        class="icon"
      >
        <a
          href="#"
        >
          <i
            class="fa fa-bars"
          />
        </a>
      </li>
    </ul>
  </nav>
</DocumentFragment>
`;
